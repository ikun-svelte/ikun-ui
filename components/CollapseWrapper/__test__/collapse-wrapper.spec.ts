import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KCollapseWrapper from '../src';

let host: HTMLElement;

const initHost = () => {
	host = document.createElement('div');
	host.setAttribute('id', 'host');
	document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KCollapseWrapper', () => {
	test('props: cls', async () => {
		const instance = new KCollapseWrapper({
			target: host,
			props: {
				cls: 'k-collapse-wrapper--test'
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-collapse-wrapper--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
