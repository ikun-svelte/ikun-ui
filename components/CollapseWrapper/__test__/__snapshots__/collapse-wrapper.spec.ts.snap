// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KCollapseWrapper > props: cls 1`] = `"<div class=\\"k-collapse-wrapper k-collapse-wrapper--base k-collapse-wrapper--test\\"></div>"`;
